be/ut/dtm0_log_ut.c

```diff
@@ -0,0 +1,192 @@
+/* -*- C -*- */
+/*
+ * Copyright (c) 2013-2020 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+#define M0_TRACE_SUBSYSTEM M0_TRACE_SUBSYS_DTM
+#include "be/dtm0_log.h"
+#include "be/list.h"
+#include "dtm0/clk_src.h"
+#include "lib/errno.h"
+#include "lib/misc.h"		/* m0_forall */
+#include "ut/ut.h"		/* M0_UT_ASSERT */
+
+#define UT_DTM0_LOG_MAX_PA            3
+#define UT_DTM0_LOG_MAX_LOG_REC      10
+#define UT_DTM0_LOG_BUF_SIZE        256
+
+int ut_dl_init_pa(struct m0_dtm0_tx_pa *pa, int rand)
+{
+	m0_fid_set(&pa->pa_fid, rand + 1, rand + 1);
+	pa->pa_state = M0_DTPS_INPROGRESS;
+	return 0;
+}
+
+bool ut_dl_verify_pa(struct m0_dtm0_tx_pa *pa, int rand)
+{
+	struct m0_fid temp_fid;
+	m0_fid_set(&temp_fid, rand + 1, rand + 1);
+
+	return !m0_fid_cmp(&pa->pa_fid, &temp_fid) &&
+               pa->pa_state >= 0                   &&
+               pa->pa_state < M0_DTPS_NR;
+}
+
+void ut_dl_init_tid(struct m0_dtm0_tid *tid, int rand)
+{
+	m0_fid_set(&tid->dti_fid, rand + 1, rand + 1);
+	tid->dti_ts.dts_phys = rand + 1;
+}
+
+bool ut_dl_verify_tid(struct m0_dtm0_tid *tid, int rand)
+{
+	struct m0_fid temp_fid;
+	m0_fid_set(&temp_fid, rand + 1, rand + 1);
+
+	return !m0_fid_cmp(&tid->dti_fid, &temp_fid) &&
+               (tid->dti_ts.dts_phys == rand + 1);
+}
+
+void ut_dl_init_txd(struct m0_dtm0_tx_desc *txd, int rand)
+{
+	m0_dtm0_tx_desc_init(txd, UT_DTM0_LOG_MAX_PA);
+	ut_dl_init_tid(&txd->dtd_id, rand);
+
+	m0_forall(i, txd->dtd_pg.dtpg_nr,
+                  !ut_dl_init_pa(&txd->dtd_pg.dtpg_pa[i], rand));
+}
+
+bool ut_dl_verify_txd(struct m0_dtm0_tx_desc *txd, int rand)
+{
+	return ut_dl_verify_tid(&txd->dtd_id, rand)        &&
+               (txd->dtd_pg.dtpg_nr == UT_DTM0_LOG_MAX_PA) &&
+	       m0_forall(i, txd->dtd_pg.dtpg_nr,
+                         ut_dl_verify_pa(&txd->dtd_pg.dtpg_pa[i], rand));
+}
+
+void ut_dl_init_buf(struct m0_buf *buf, int rand)
+{
+	memset(buf, 0, sizeof(*buf));
```

Comment:
`M0_SET0(buf)` can be used instead.

Reply:
Done at 95c4570f119e065c3e5ba9c2c214a8f6f9a83093